### PR TITLE
Added authentication

### PR DESCRIPTION
**Pull request by Ishan27g**
- Root api endpoint updated to login/signup page
- Backend + DB re-used from my exisiting project, deployed to Heroku.
- `package.json` updated to compile `auth.ts`

### CODE REVIEW

File: src/server.ts [hunk 1]
```diff
@@ -17,7 +17,32 @@ const httpServer = http.createServer(app);
...
+  return res.sendFile(path.join(__dirname, "client", "auth.html"));
+});
+
+app.get("/join", async (req, res) => {
+    const token = req.query.token;
+    if (!token) {
+      return res.redirect("/");
+    }
+    return res.sendFile(path.join(__dirname, "client", "join.html"));
+    /*
```
Comment (dalyIsaac) [inline src/server.ts:29]: I'm assuming this is meant to be commented out...

**Review by dalyIsaac — APPROVED** [review]
LGTM

File: src/server.ts (hunk 1 above)
Comment (Ishan27g) [reply src/server.ts:29]: No, I think its correct. If the token param isn't present then redirect to homepage, else forward to the create/join page.

auth.html will forward the valid request to /join

**Comment by dalyIsaac** [comment]
Wait, did you mean to target the branch `no-framework`? Or did you mean to merge into `main`?

**Pull request merged**